### ISSUE TITLE
Provide database integration tests best practices

### ISSUE THREAD

**Issue by 0xTim** [vapor/docs#594]
We should add a section in the docs for best practices and options when running integration tests with a real database.

Points to touch on should include:

* setting up the database configuration for testing (you don't want to use the live DB!)
* using `autoRevert()`/`autoMigrate()` in tests
* using a transaction in tests to speed them up

**Comment by alemohamad** [comment]
@0xTim  with the new merged PR, can this be closed? Thanks!

**Comment by 0xTim** [comment]
Yes lets close for now and can create separate issues if there's demand
